contracts/5/IbETHRouter.sol

```diff
@@ -590,59 +645,61 @@ contract UniswapV2Router02 is IUniswapV2Router02 {
         require(
             IERC20(path[path.length - 1]).balanceOf(to).sub(balanceBefore) >=
                 amountOutMin,
-            "UniswapV2Router: INSUFFICIENT_OUTPUT_AMOUNT"
+            "IbETHRouter: INSUFFICIENT_OUTPUT_AMOUNT"
         );
     }
 
+    // Same as swapExactETHForTokens, just with supporting fee on transfer.
     function swapExactETHForTokensSupportingFeeOnTransferTokens(
         uint256 amountOutMin,
         address[] calldata path,
         address to,
         uint256 deadline
     ) external payable ensure(deadline) {
-        require(path[0] == WETH, "UniswapV2Router: INVALID_PATH");
-        uint256 amountIn = msg.value;
-        IWETH(WETH).deposit.value(amountIn)();
+        require(path[0] == ibETH, "IbETHRouter: INVALID_PATH");
+        uint256 amountIn = msg.value;                
+        Bank(ibETH).deposit.value(amountIn)();        
         assert(
-            IWETH(WETH).transfer(
+            Bank(ibETH).transfer(
                 UniswapV2Library.pairFor(factory, path[0], path[1]),
-                amountIn
+                ETHToIbETH(amountIn)
             )
         );
         uint256 balanceBefore = IERC20(path[path.length - 1]).balanceOf(to);
-        _swapSupportingFeeOnTransferTokens(path, to);
+        _swapSupportingFeeOnTransferTokens(path, to);        
         require(
             IERC20(path[path.length - 1]).balanceOf(to).sub(balanceBefore) >=
                 amountOutMin,
-            "UniswapV2Router: INSUFFICIENT_OUTPUT_AMOUNT"
+            "IbETHRouter: INSUFFICIENT_OUTPUT_AMOUNT"
         );
     }
 
+    // Same as swapExactTokensForETH, just with supporting fee on transfer.
     function swapExactTokensForETHSupportingFeeOnTransferTokens(
         uint256 amountIn,
         uint256 amountOutMin,
         address[] calldata path,
         address to,
         uint256 deadline
     ) external ensure(deadline) {
-        require(path[path.length - 1] == WETH, "UniswapV2Router: INVALID_PATH");
+        require(path[path.length - 1] == ibETH, "IbETHRouter: INVALID_PATH");
         TransferHelper.safeTransferFrom(
             path[0],
             msg.sender,
             UniswapV2Library.pairFor(factory, path[0], path[1]),
             amountIn
         );
         _swapSupportingFeeOnTransferTokens(path, address(this));
-        uint256 amountOut = IERC20(WETH).balanceOf(address(this));
+        uint256 amountOut = Bank(ibETH).balanceOf(address(this));
         require(
             amountOut >= amountOutMin,
-            "UniswapV2Router: INSUFFICIENT_OUTPUT_AMOUNT"
-        );
-        IWETH(WETH).withdraw(amountOut);
-        TransferHelper.safeTransferETH(to, amountOut);
-    }
+            "IbETHRouter: INSUFFICIENT_OUTPUT_AMOUNT"
+        );                
+        Bank(ibETH).withdraw(amountOut);        
+        TransferHelper.safeTransferETH(to, ExactIbETHToETH(amountOut));        
```

Comment:
After we withdraw, this number may not be the same. We should compute ethAmountOut before doing withdraw